### PR TITLE
Enable auto discovery of agents

### PR DESCRIPTION
**Pull request by abhi1287**
Enable auto discovery of agents

### CODE REVIEW

File: alica_ros_proxy/src/communication/AlicaRosCommunication.cpp [hunk 1]
```diff
@@ -305,11 +317,72 @@ void AlicaRosCommunication::handleSolverResult(const alica_msgs::SolverResult& s
...
+void AlicaRosCommunication::sendAgentAnnouncement(const AgentAnnouncement& pa) const
+{
+    alica_msgs::AgentAnnouncement newpa;
+    newpa.sender_id.id = pa.senderID->toByteVector();
+    newpa.sender_name = pa.senderName;
+    newpa.sender_sdk = pa.senderSdk;
+    newpa.plan_hash = pa.planHash;
+    newpa.role = pa.role;
+    for (const auto& cap : pa.capabilities) {
+        // alica_msgs::StringTuple()
```
Comment (abolinsky) [inline alica_ros_proxy/src/communication/AlicaRosCommunication.cpp]: remove comment

File: supplementary_tests/CMakeLists.txt [hunk 2]
```diff
@@ -1,10 +1,8 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(supplementary_tests)
 
-## Use c++ 11x std
-set(CMAKE_CXX_FLAGS "-std=c++11 ${CMAKE_CXX_FLAGS}")
 if (${CMAKE_EXTRA_GENERATOR} MATCHES "Eclipse CDT4")
-	set(CMAKE_CXX_COMPILER_ARG1 "-std=c++11" CACHE STRING "C++ version for eclipse" FORCE)
+	set(CMAKE_CXX_COMPILER_ARG1 "-std=c++14" CACHE STRING "C++ version for eclipse" FORCE)
```
Comment (abolinsky) [inline supplementary_tests/CMakeLists.txt:5]: If you want to enforce c++14, this should be set regardless of which IDE is being used.

**Review by abolinsky — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -16,7 +16,7 @@ env:
     - COVERITY_EMAIL=alexander.bolinsky@rapyuta-robotics.com
     - PYTHONPATH=$PYTHONPATH:/usr/lib/python2.7/dist-packages:/usr/local/lib/python2.7/dist-packages
     - BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi)
-    - BASE_BRANCH=rr-devel
+    - BASE_BRANCH=$TRAVIS_BRANCH
     - TRAVIS_BRANCH=$TRAVIS_BRANCH
```
Comment (abolinsky) [inline .travis.yml:20]: Can remove TRAVIS_BRANCH

**Review by abolinsky — CHANGES_REQUESTED** [review]

File: supplementary_tests/CMakeLists.txt (hunk 2 above)
Comment (abhi1287) [reply supplementary_tests/CMakeLists.txt:5]: That's done anyways in common config.

**Comment by abhi1287** [comment]
Extension of auto-discovery PR in the main alica repo: https://github.com/rapyuta-robotics/alica/pull/66

File: supplementary_tests/CMakeLists.txt (hunk 2 above)
Comment (abolinsky) [reply supplementary_tests/CMakeLists.txt:5]: Then do we need this line?

File: supplementary_tests/src/test/etc/Globals.conf [hunk 4]
```diff
@@ -1,111 +0,0 @@
...
-			IsGoalie = False
-			Speed = Fast
-			CanPass = True
-			LocalizationSuccess = -1
-			#DefaultRole = G10Attacker
-			DefaultRole = Attacker
-		[!Brain]
-	[!Team]
-	
-	[RolePriority]
```
Comment (gauthamm) [inline supplementary_tests/src/test/etc/Globals.conf:104]: Is role priority not being used?

File: supplementary_tests/src/test/etc/Globals.conf [hunk 5]
```diff
@@ -1,111 +0,0 @@
-[Globals]
-
-	#OwnTeamColour = cyan
-	OwnTeamColour = magenta
```
Comment (gauthamm) [inline supplementary_tests/src/test/etc/Globals.conf:4]: We can still have globals.conf for setting common parameters if they are independent of the agent.

**Review by gauthamm — APPROVED** [review]

**Review by abolinsky — APPROVED** [review]

File: supplementary_tests/src/test/etc/Globals.conf (hunk 4 above)
Comment (abhi1287) [reply supplementary_tests/src/test/etc/Globals.conf:104]: not at the moment.

File: supplementary_tests/src/test/etc/Globals.conf (hunk 5 above)
Comment (abhi1287) [reply supplementary_tests/src/test/etc/Globals.conf:4]: Makes sense but let's do it in when someone asks for that feature. Capabilities are not being used anyways in the engine for now.

**Pull request merged**